File: 125_algorithms/_exercises/templates/_algorithms_challenges/leetcode/leetCode/Stack/225_ImplementStackusingQueues.cpp
/*
 * @Author: xuezaigds@gmail.com
 * @Last Modified time: 2016-04-24 13:24:37
 */

c.. Stack {
public:
    // Push element x onto stack.
    // then rotating the queue until the new element is at the front
    void push(int x) {
        q.push(x);
        ___(int i=0;i<q.size()-1;i++){
            q.push(q.front());
            q.pop();
        }
    }

    // Removes the element on top of the stack.
    void pop() {
        q.pop();
    }

    // Get the top element.
    int top() {
        r_ q.front();
    }

    // Return whether the stack is empty.
    bool empty() {
        r_ q.empty();
    }
private:
    queue<int> q{};
};

File: 125_algorithms/_exercises/templates/_algorithms_challenges/leetcode/leetCode/Stack/225_ImplementStackusingQueues.py
#! /usr/bin/env python
# -*- coding: utf-8 -*-
# @Author: xuezaigds@gmail.com
from collections import deque


c.. Stack o..
    ___ __init__(self
        self._queue = deque()

    ___ push  x
        # Pushing to back and
        # then rotating the queue until the new element is at the front
        q = self._queue
        q.a.. x)
        ___ i __ xrange(l..(q) - 1
            q.a.. q.popleft())

    ___ pop(self
        self._queue.popleft()

    ___ top(self
        r_ self._queue[0]

    ___ empty(self
        r_ n.. l..(self._queue)

"""Test
if __name__ == '__main__':
    s = Stack()
    s.push(1)
    s.push(2)
    print s.top()
    s.pop()
    print s.empty()
    print s.top()
    s.pop()
    print s.empty()
"""
